File: typescript/merge.ts
function merge(intervals: number[][]): number[][] {
  let il = intervals.length;
  let ans: number[][] = new Array();
  intervals.sort((a: number[], b: number[])=> a[0]-b[0]);
  for (const i of intervals) {
    if (ans.length > 0 && i[0]<=ans[ans.length-1][1]) {
      ans[ans.length-1][1] = Math.max(ans[ans.length-1][1], i[1]);
    } else {
      ans.push(i);
    }
  }
  return ans;  
};

File: rust/questions/merge.rs
impl Solution {
    pub fn merge(mut intervals: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        intervals.sort_unstable_by_key(|v| v[0]);
        let mut ans: Vec<Vec<i32>> = Vec::new();
        for cur in intervals {
            match ans.last_mut() {
                Some(last) if cur[0] <= last[1] => {
                    last[1] = last[1].max(cur[1]);
                }
                _ => ans.push(cur),
            }
        }
        ans
    }
}
